akka-testkit-typed/src/main/scala/akka/testkit/typed/BehaviorTestkit.scala

```diff
@@ -172,8 +172,17 @@ class BehaviorTestkit[T] private (_name: String, _initialBehavior: Behavior[T])
   }
 
   private var current = Behavior.validateAsInitial(Behavior.undefer(_initialBehavior, ctx))
+  private var currentUncanonical = current
 
+  /**
+   * Returns the current canonical behavior
+   */
   def currentBehavior: Behavior[T] = current
+
+  /**
+   * Returns the current behavior before being converted to the canonical one
```

Comment:
I don't think users will have a clue of what canonical is.
What about: Returns the current behavior as it was returned from processing previous message with `run`. For example if `Behaviors.unhandled` is returned that will be kept here, but not in `[[currentBehavior]]`.